core/src/test/java/com/yahoo/oak/ComputeTest.java

```diff
@@ -116,14 +127,18 @@ public void testMain() throws InterruptedException {
             oak.zc().putIfAbsent(key, val);
         }
 
-        for (int i = 0; i < NUM_THREADS; i++) {
-            threads.get(i).start();
-        }
-
         latch.countDown();
 
-        for (int i = 0; i < NUM_THREADS; i++) {
-            threads.get(i).join();
+        executor.shutdown();
+        try {
+            executor.shutdown();
+            if (!executor.awaitTermination(10, TimeUnit.SECONDS)) {
+                executor.shutdownNow();
+                Assert.fail("should have done all the tasks in time");
```

Comment:
```suggestion
                Assert.fail("Failed to finish all tasks in time");
```

Reply:
The await termination was removed so that not needed